tests/scripts/all.sh

```diff
@@ -321,6 +322,12 @@ else
     fi
 fi
 
+if [ $RUN_ARMCC -eq 0 ] && [ $YOTTA -ne 0 ]; then
```

Comment:
Do we really want to fail on this error? What about `--keep-going`?

Reply:
I'd say we want to fail here because:
- this is a user error, not a failing test
- the error is reported early, before any test is run
- this is consistent with how we handle environment checks (which are likely user errors too) just above.

Reply:
Keep-going mode is designed to keep going if there's an expected failure: compilation error, failed test. It isn't designed to keep going in the face of an unexpected failure such as bad command line options or a badly set up environment. So exiting in keep-going mode here is fine.